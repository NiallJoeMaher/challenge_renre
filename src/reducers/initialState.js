export default {
  forecast: [],
  hasErrored: false,
  isLoading: false,
  unit: 'c',
  location: 'Dublin',
};
