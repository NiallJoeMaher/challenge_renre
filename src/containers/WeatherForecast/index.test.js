import React from 'react';
import { shallow } from 'enzyme';

import { WeatherForecast } from './';

const mockForecastArray = [{
    "code": "39",
    "date": "30 Nov 2018",
    "day": "Fri",
    "high": "7",
    "low": "5",
    "text": "Scattered Showers"
  },
  {
    "code": "39",
    "date": "1 Dec 2018",
    "day": "Sat",
    "high": "7",
    "low": "5",
    "text": "Scattered Showers"
  },
  {
    "code": "29",
    "date": "2 Dec 2018",
    "day": "Sun",
    "high": "7",
    "low": "5",
    "text": "Scattered Showers"
  },
  {
    "code": "39",
    "date": "3 Dec 2018",
    "day": "Mon",
    "high": "7",
    "low": "5",
    "text": "Scattered Showers"
  },

]

import initialState from '../../reducers/initialState';
it('renders as expected with default params', () => {
  const component = shallow(
    <WeatherForecast {...initialState} fetchForecast={() => {}} />
  );
  expect(component).toMatchSnapshot();
});

it('renders as expected with forecast params', () => {
  const component = shallow(
    <WeatherForecast  
      {...initialState}
      forecast={mockForecastArray}
      fetchForecast={() => {}} 
    />
  );
  expect(component).toMatchSnapshot();
});

it('renders as expected with error', () => {
  const component = shallow(
    <WeatherForecast
      {...initialState}
      fetchForecast={() => {}}
      hasErrored={true}
    />
  );
  expect(component).toMatchSnapshot();
});
