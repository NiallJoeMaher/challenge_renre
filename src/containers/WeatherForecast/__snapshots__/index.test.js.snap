// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected with default params 1`] = `
<div
  className="c-WeatherForecast"
>
  <div
    className="c-WeatherForecast__header"
  >
    RenRe Weather App
  </div>
  <div
    className="c-WeatherForecast__body"
  >
    <div
      className="c-WeatherForecast__no-card-data"
    >
       
      Checking our satelites!
    </div>
    <Switch
      leftLabel="째C"
      on={false}
      onClick={[Function]}
      rightLabel="째F"
    />
  </div>
  <div
    className="c-WeatherForecast__button-group"
  >
    <Button
      active={true}
      key="Dublin"
      onClick={[Function]}
      text="Dublin"
    />
    <Button
      active={false}
      key="London"
      onClick={[Function]}
      text="London"
    />
    <Button
      active={false}
      key="New York"
      onClick={[Function]}
      text="New York"
    />
  </div>
</div>
`;

exports[`renders as expected with error 1`] = `
<div
  className="c-WeatherForecast__error"
>
   
  Error occured, please try again...
   
</div>
`;

exports[`renders as expected with forecast params 1`] = `
<div
  className="c-WeatherForecast"
>
  <div
    className="c-WeatherForecast__header"
  >
    RenRe Weather App
  </div>
  <div
    className="c-WeatherForecast__body"
  >
    <div
      className="c-WeatherForecast__card-group"
    >
      <WeatherCard
        date="30 Nov 2018"
        day="Fri"
        highValue="7"
        key="30 Nov 2018"
        label="Fri"
        lowValue="5"
      />
      <WeatherCard
        date="1 Dec 2018"
        day="Sat"
        highValue="7"
        key="1 Dec 2018"
        label="Sat"
        lowValue="5"
      />
      <WeatherCard
        date="2 Dec 2018"
        day="Sun"
        highValue="7"
        key="2 Dec 2018"
        label="Sun"
        lowValue="5"
      />
    </div>
    <Switch
      leftLabel="째C"
      on={false}
      onClick={[Function]}
      rightLabel="째F"
    />
  </div>
  <div
    className="c-WeatherForecast__button-group"
  >
    <Button
      active={true}
      key="Dublin"
      onClick={[Function]}
      text="Dublin"
    />
    <Button
      active={false}
      key="London"
      onClick={[Function]}
      text="London"
    />
    <Button
      active={false}
      key="New York"
      onClick={[Function]}
      text="New York"
    />
  </div>
</div>
`;
